src/Avalonia.Controls/TextBox.cs

```diff
@@ -308,13 +309,20 @@ protected override void OnLostFocus(RoutedEventArgs e)
             _presenter?.HideCaret();
         }
 
-        protected override void OnTextInput(TextInputEventArgs e)
+        protected override void OnTextInputHandlerSelection(TextInputHandlerSelectionEventArgs e)
```

Comment:
What scenario does this cover that simply trying to cast the control to `ITextInputHandler` doesn't cover? Is it to allow a user-level override of the text input handler, i.e. someone writing an application could choose another `ITextInputHandler` for a `TextBox`?

If so, what other handlers do you forsee for this?

Reply:
>simply trying to cast the control

*What* control exactly? We could try to check for the currently focused control, but I'd like to have a way for more complex scenario. E. g. some kind of inline completion list taking focus for handling arrow keys while delegating text input to the parent control